### PR TITLE
fix: debug_traceTransaction is not working with popular tools

### LINKED ISSUE #2777: debug_traceTransaction is not working with popular tools
### Description

The current parameters that we accept for the `debug_traceTransaction` endpoint are inconsistent with what other providers expect. Also debugging transactions in Remix is not working currently because of that.

**References:**
- https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-debug#debugtracetransaction
- https://docs.chainstack.com/reference/ethereum-tracetransaction
- https://docs.infura.io/api/networks/linea/json-rpc-methods/debug/debug_tracetransaction
- https://docs.alchemy.com/reference/debug-tracetransaction
- https://www.quicknode.com/docs/ethereum/debug_traceTransaction

### Steps to reproduce

1. Go to remix
2. Deploy a contract
3. Call a contract function
4. Find the transaction hash of the function call
5. Go to the debugger
6. Enter the transaction hash
7. The remix console logs a validation error from our JSON-RPC relay

### Additional context

_No response_

### Hedera network

_No response_

### Version

v0.52.1

### Operating system

None

### PR DESCRIPTION
**Pull request by victor-yanev**
**Description**:

The current parameters that we accept for the `debug_traceTransaction` endpoint are inconsistent with what other providers expect. 

Also debugging transactions in Remix is not working currently because of that.

**References:**

https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-debug#debugtracetransaction

https://docs.chainstack.com/reference/ethereum-tracetransaction

https://docs.infura.io/api/networks/linea/json-rpc-methods/debug/debug_tracetransaction

https://docs.alchemy.com/reference/debug-tracetransaction

https://www.quicknode.com/docs/ethereum/debug_traceTransaction

**Related issue(s)**:

Fixes #2777 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Manually tested through Remix:
  - [x] Tracked what parameters are received from Remix on clicking the "Debug" button:
    image
  - [x] Successfully used the built-in Debugger Tool in Remix after applying the changes in the code:
    Screenshot 2024-08-02 at 12 01 10

### CODE REVIEW

File: packages/relay/src/lib/services/debugService/index.ts [hunk 1]
```diff
@@ -22,13 +22,14 @@ import type { Logger } from 'pino';
...
-import { decodeErrorMessage, numberTo0x, strip0x } from '../../../formatters';
+import { decodeErrorMessage, mapKeysAndValues, numberTo0x, strip0x } from '../../../formatters';
 import constants from '../../constants';
 import { TracerType, CallType } from '../../constants';
 import { predefined } from '../../errors/JsonRpcError';
 import { EthImpl } from '../../eth';
 import { IOpcodesResponse } from '../../clients/models/IOpcodesResponse';
 import { IOpcode } from '../../clients/models/IOpcode';
+import { ICallTracerConfig, IOpcodeLoggerConfig, ITracerConfig } from '../../types/ITracerConfig';
 const SUCCESS = 'SUCCESS';
```
Comment (quiet-node) [inline packages/relay/src/lib/services/debugService/index.ts:33]: This can go to constants.ts

**Review by quiet-node — CHANGES_REQUESTED** [review]
Great work! Just a couple of suggestions

File: packages/relay/src/lib/services/debugService/IDebugService.ts [hunk 2]
```diff
@@ -19,12 +19,13 @@
  */
```
[the remark this answers is not in the document]
Comment (victor-yanev) [reply packages/relay/src/lib/services/debugService/IDebugService.ts:19]: The `IDebugService` interface is meant to define the contract for the debug service class, outlining the methods that need to be implemented by the service class.

On the other hand, `ITracerConfig` and `ITracerConfigWrapper` define the configuration objects for tracing. While these configurations are used by the debug service, they are conceptually distinct and they shouldn't be placed in the same file.

Btw, service interfaces are placed in the same directory as their service implementations throughout the project. So, it makes sense to stick with this pattern here too.

**Review by acuarica — DISMISSED** [review]
lg, left some questions

File: packages/relay/tests/lib/eth/eth_estimateGas.spec.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ import { numberTo0x } from '../../../src/formatters';
 import { Eth, JsonRpcError } from '../../../src';
 import { generateEthTestEnv } from './eth-helpers';
 import { Precheck } from '../../../src/lib/precheck';
-import { createStubInstance, stub, SinonStub, SinonStubbedInstance } from 'sinon';
+import { createStubInstance, SinonStub, SinonStubbedInstance, stub } from 'sinon';
```
[the remark this answers is not in the document]
Comment (victor-yanev) [reply packages/relay/tests/lib/eth/eth_estimateGas.spec.ts:32]: Yes, sorry about that! 

I forgot to turn off the "Optimize imports" option in my IDE when I was doing a merge commit from the main branch and I clicked on cancel, but it already had processed some files. 😅

**Review by jeromy-cannon — APPROVED** [review]

**Review by acuarica — DISMISSED** [review]
lg

**Review by konstantinabl — APPROVED** [review]

**Review by quiet-node — APPROVED** [review]

**Pull request merged**